dev/breeze/src/airflow_breeze/ci/build_image.py

```diff
@@ -0,0 +1,93 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# from airflow_breeze.utils import run_command
+import os
+import shlex
+
+from airflow_breeze.breeze import get_airflow_sources_root
+from airflow_breeze.cache import check_cache_and_write_if_not_cached
+from airflow_breeze.ci.build_params import BuildParams
+from airflow_breeze.utils import run_command
+
+PARAMS_CI_IMAGE = [
+    "python_base_image",
+    "airflow_version",
+    "airflow_branch",
+    "airflow_extras",
+    "airflow_pre_cached_pip_packages",
+    "additional_airflow_extras",
+    "additional_python_deps",
+    "additional_dev_apt_command",
+    "additional_dev_apt_deps",
+    "additional_dev_apt_env",
+    "additional_runtime_apt_command",
+    "additional_runtime_apt_deps",
+    "additional_runtime_apt_env",
+    "upgrade_to_newer_dependencies",
+    "constraints_github_repository",
+    "airflow_constraints_reference",
+    "airflow_constraints",
+    "airflow_image_repository",
+    "airflow_image_date_created",
+    "build_id",
+    "commit_sha",
+]
+
+PARAMS_TO_VERIFY_CI_IMAGE = [
+    "dev_apt_command",
+    "dev_apt_deps",
+    "runtime_apt_command",
+    "runtime_apt_deps",
+]
+
+
+def construct_arguments_docker_command(ci_image: BuildParams) -> str:
+    args_command = ""
+    for param in PARAMS_CI_IMAGE:
+        args_command += " --build-arg " + param.upper() + "=" + str(getattr(ci_image, param))
+    for verify_param in PARAMS_TO_VERIFY_CI_IMAGE:
+        param_value = str(getattr(ci_image, verify_param))
+        if len(param_value) > 0:
+            args_command += " --build-arg " + verify_param.upper() + "=" + param_value
+    args_command += " " + ci_image.docker_cache_ci_directive
```

Comment:
```suggestion
    args_command = []
    for param in PARAMS_CI_IMAGE:
        args_command += ["--build-arg", f"{param.upper()}={str(getattr(ci_image, param))}]
    for verify_param in PARAMS_TO_VERIFY_CI_IMAGE:
        param_value = str(getattr(ci_image, verify_param))
        if len(param_value) > 0:
            args_command += ["--build-arg", f"{verify_param.upper()}={param_value}"]
    args_command += [*ci_image.docker_cache_ci_directive]
```

Reply:
Is there any specific reason to follow the above way of implementation @mik-laj ? Could you explain me about it?

Reply:
I get an error when I tried the above way. That's why i was wondering if there is any specific reason to follow this way

Reply:
This is all about "space" treatment. 

If you want to pass argument with space (say "arg with space" ) -you have surround it with `"`  or `'` and it gets more complex (and platform dependente) as you also want to include '"". If you have arg with space and `"` is not really obvious how to pass. 

so rather than `command arg1 "arg with space"`  it's much better to specify the args as array:

```
["command", "arg1", "arg with space"] 
```

Reply:
Additional info: See "note" here: https://docs.python.org/3/library/subprocess.html#popen-constructor  it tells:

> args should be a sequence of program arguments or else a single string or path-like object. By default, the program to execute is the first item in args if args is a sequence. If args is a string, the interpretation is platform-dependent and described below. See the shell and executable arguments for additional differences from the default behavior. Unless otherwise stated, it is recommended to pass args as a sequence.

Notes below explain it in detail.